smithy-aws-traits/src/main/java/software/amazon/smithy/aws/traits/tagging/TaggingShapeUtils.java

```diff
@@ -190,10 +190,10 @@ static boolean isTagPropertyInInput(
         Model model,
         ResourceShape resource
     ) {
-        PropertyBindingIndex propertyBindingIndex = PropertyBindingIndex.of(model);
-        Optional<String> property = resource.expectTrait(TaggableTrait.class).getProperty();
-        if (property.isPresent()) {
-            if (operationId.isPresent()) {
+        if (operationId.isPresent()) {
```

Comment:
The only way for this code to branch to `true` is for when the optional `operationId` is present. I moved it up to avoid unneeded work when that's not true.